STIPS/STIP-009.md

```diff
@@ -0,0 +1,1098 @@
+        _extensions
+    );
+    ```
+
++ set temporary initialization metadata for the newly created SetToken and DelegatedManager
+    ```solidity
+    initialize[setTokenAddress] = InitializeParams({
+        deployer: msg.sender,
+        owner: _owner,
+        manager: managerAddress,
+        isPending: true
+    });
+    ```
+
+ANYONE CAN CALL: Deploys a DelegatedManager and sets some temporary metadata about
+the deployment which will be read during a subsequent intialization step which wires everything
+together. This method is used when migrating an existing SetToken to the DelegatedManager system
+
+> createManager
+
+```solidity
+function createManager(
+    address memory _setTokenAddress,
+    address _owner,
+    address _methodologist,
+    address[] memory _modules,
+    address[] memory _operators,
+    address[] memory _assets,
+    address[] memory _extensions
+)
+```
+
++ if `_assets` are specified, require that all of the sets existing *_components* are included in the *_assets* array
+
++ deploy the manager, setting its *useAssetsAllowedList* constructor parameter to true if there are *_assets* and false otherwise.
+    ```solidity
+    address managerAddress = _deployManager(
+        setTokenAddress,
+        address(this),
+        _methodologist,
+        useAssetsAllowedList
+        _operators,
+        _assets,
+        _extensions
+    );
+    ```
+
++ set temporary initialization metadata for the newly created SetToken and DelegatedManager
+    ```solidity
+    initialize[setTokenAddress] = InitializeParams({
+        deployer: msg.sender,
+        owner: _owner,
+        manager: managerAddress,
+        isPending: true
+    });
+    ```
+
+> initialize
+
+ANYONE CAN CALL: Wires SetToken, DelegatedManager, global manager extensions, and modules together into
+a functioning package. `_initializeTargets` includes any extensions or modules which need to be initialized
+and modules must initialized before extensions.`initializeBytecode` is an encoded call to the
+relevant target's initialize function.
+
+```solidity
+function initialize(
+    address memory _setTokenAddress,
+    uint256 _ownerFeeSplit,
+    address _ownerFeeRecipient,
+    address[] memory _initializeTargets,
+    bytes[] memory _initializeBytecode,
+)
+```
++ require that caller be the *deployer* specified in the *initialize[_setTokenAddress]* mapping
++ require that *initialize[_setTokenAddress].isPending* is `true`
++ require that *_initializeExtensionTargets* and *initializeExtensionBytecode* arrays have same length
++ call DelegatedManager.setOwnerFeeSplit with *_ownerFeeSplit*
++ call DelegatedManager.setOwnerFeeRecipient with *_ownerFeeRecipient*
+
++ for each target, bytecode in _initializeTargets, _initializeBytecode
+    + call target.functionCallWithValue(bytecode, 0)
+
++ if setToken manager is this factory we're creating a new SetToken rather than migrating
+  + call `setToken.setManager(initialize[_setTokenAddress].manager)`
```

Comment:
The manager contract needs to be the `manager` before this function is called. So this step needs to happen in the create function. In this function though we should set the `owner` (which we do on the next line).

Reply:
We discussed this offline and think it's ok for the factory to be the owner during the initialization sequence. Migrating tokens will need to call `setManager` in a separate final step.